stock_intercompany/models/stock_picking.py

```diff
@@ -41,7 +41,7 @@ def _get_counterpart_picking_vals(self, company):
             "picking_type_id": ptype.id,
             "state": "draft",
             "location_id": self.env.ref("stock.stock_location_suppliers").id,
-            "location_dest_id": warehouse.lot_stock_id.id,
+            "location_dest_id": ptype.default_location_dest_id.id,
```

Comment:
we should still default to warehouse main stock location in case default loc on operation type isn't specified
```suggestion
            "location_dest_id": ptype.default_location_dest_id.id or warehouse.lot_stock_id.id,
```